lib/github_changelog_generator/generator/generator_tags.rb

```diff
@@ -2,11 +2,33 @@ module GitHubChangelogGenerator
   class Generator
     # fetch, filter tags, fetch dates and sort them in time order
     def fetch_and_filter_tags
```

Comment:
The logic should be pretty clear now. The steps are very serial.

1. detect the since_tag and due_tag up front, before any logic
2. then call fetcher to get all tags
3. then fetch tag dates
4. then sort by date
5. then filter tags
6. then build tag section mappings